ctapipe/io/hdf5eventsource.py

```diff
@@ -204,6 +204,9 @@ def __init__(self, input_url=None, config=None, parent=None, **kwargs):
 
         version = self.file_.root._v_attrs["CTA PRODUCT DATA MODEL VERSION"]
         self.datamodel_version = tuple(map(int, version.lstrip("v").split(".")))
+        self._obs_ids = tuple(
```

Comment:
Not really. The provenance is written as last step, after the exit stack calls all the cleanups. It accesses the processed obs ids, which was not possible since the file was already closed.

Reply:
So the change is directly related to the rest